### PR TITLE
fix: use enumerator to validate kafka connection

### LINKED ISSUE #11315: `CREATE SINK` succeeds for invalid kafka config
### Describe the bug

There are error logs, but the statement succeeds.

```
2023-07-28T23:14:09.832076+02:00 ERROR librdkafka: librdkafka: FAIL [thrd:absolutely-invalid:9092/bootstrap]: absolutely-invalid:9092/bootstrap: Failed to resolve 'absolutely-invalid:9092': nodename nor servname provided, or not known (after 2ms in state CONNECT, 17 identical error(s) suppressed)
2023-07-28T23:14:09.832218+02:00 ERROR madsim_rdkafka::std_::client: librdkafka: Global error: Resolve (Local: Host resolution failure): absolutely-invalid:9092/bootstrap: Failed to resolve 'absolutely-invalid:9092': nodename nor servname provided, or not known (after 2ms in state CONNECT, 17 identical error(s) suppressed)
```

### Error message/log

_No response_

### To Reproduce

```sql
dev=> create table tt(x int ) append only;
NOTICE:  APPEND ONLY TABLE is currently an experimental feature.
CREATE_TABLE
dev=> create sink s from tt with(connector='kafka', type='append-only',  properties.bootstrap.server='absolutely-invalid',topic='a');
CREATE_SINK
```

### Expected behavior

_No response_

### How did you deploy RisingWave?

_No response_

### The version of RisingWave

latest main

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tabVersion**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

resolve #11315

## Checklist

- [ ] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: scripts/source/test_data/test-rw-sink-append-only.1
Comment (StrikeW) [inline scripts/source/test_data/test-rw-sink-append-only.1:1]: Why add these empty files?

File: src/connector/src/sink/kafka.rs [hunk 1]
```diff
@@ -290,9 +307,13 @@ impl Sink for KafkaSink {
         }
 
         // Try Kafka connection.
-        // TODO: Reuse the conductor instance we create during validation.
-        KafkaTransactionConductor::new(self.config.clone(), &"validation".to_string()).await?;
-
+        // There is no such interface for kafka producer to validate a connection
```
Comment (StrikeW) [inline src/connector/src/sink/kafka.rs:310]: May refactor the comment to be explicitly, e.g. "use enumerator to validate connection properties"

**Review by StrikeW — COMMENTED** [review]
Rest lgtm

File: scripts/source/test_data/test-rw-sink-append-only.1
Comment (tabVersion) [reply scripts/source/test_data/test-rw-sink-append-only.1:1]: enumerator will verify whether the broker exists and the topic exists. the empty file is used to create the corresponding topics on the broker.

**Review by StrikeW — APPROVED** [review]

**Pull request merged**